test/lib/checkevents_test.go

```diff
@@ -0,0 +1,262 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lib
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+)
+
+type dummyEventGet struct {
+	lock sync.Mutex
+
+	minSeq int
+	allEv  []EventInfo
+	clean  bool
+	tCalls int
+}
+
+func newDummyEventGet() *dummyEventGet {
+	return &dummyEventGet{minSeq: 1}
+}
+
+func (deg *dummyEventGet) getMinMax() (minRet int, maxRet int, errRet error) {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	if deg.clean {
+		panic("getminmax called on cleaned event getter")
+	}
+	min := deg.minSeq
+	max := min + len(deg.allEv) - 1
+	return min, max, nil
+}
+func (deg *dummyEventGet) getEntry(seqno int) (EventInfo, error) {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	if deg.clean {
+		panic("getentry called on cleaned event getter")
+	}
+	min := deg.minSeq
+	max := min + len(deg.allEv) - 1
+	if seqno >= min && seqno <= max {
+		return deg.allEv[seqno-deg.minSeq], nil
+	} else {
+		return EventInfo{}, fmt.Errorf("illegal get seqno: %d, range min = %d, max = %d", seqno, min, max)
+	}
+}
+func (deg *dummyEventGet) trimThrough(seqno int) error {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	deg.tCalls++
+	if deg.clean {
+		panic("trim called on cleaned event getter")
+	}
+	min := deg.minSeq
+	max := min + len(deg.allEv) - 1
+	if seqno < 0 {
+		return fmt.Errorf("Illegal negative seqno %d", seqno)
+	} else if seqno > max {
+		return fmt.Errorf("Illegal negative seqno %d > %d", seqno, max)
+	} else if seqno < min {
+		return nil
+	}
+	deg.allEv = deg.allEv[seqno-min+1:]
+	deg.minSeq = seqno + 1
+	return nil
+}
+func (deg *dummyEventGet) trimCalls() int {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	return deg.tCalls
+}
+func (deg *dummyEventGet) cleanup() {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	if deg.clean {
+		panic("Unexpected clean")
+	} else {
+		deg.clean = true
+	}
+}
+
+func makeEvents() []EventInfo {
+	var allEv []EventInfo
+	for i := 0; i < 30; i++ {
+		ce := cloudevents.NewEvent(cloudevents.VersionV1)
+		ce.SetType("knative.dev.test.event.a")
+		ce.SetSource("https://source.test.event.knative.dev/foo")
+		ce.SetID(strconv.FormatInt(int64(i), 10))
+		allEv = append(allEv, EventInfo{Event: &ce})
+	}
+	return allEv
+}
+
+func checkEvIDEqual(t *testing.T, seen []EventInfo, expected []EventInfo) {
+	if len(seen) != len(expected) {
+		t.Fatalf("Seen ev list length %d does not match expected %d", len(seen), len(expected))
+
+	}
+	for i := range seen {
+		if seen[i].Event.ID() != expected[i].Event.ID() {
+			t.Fatalf("Id entry %d mismatch: seen %s != expected %s",
+				i, seen[i].Event.ID(), expected[i].Event.ID())
+		}
+	}
+}
+
+func (deg *dummyEventGet) setEv(firstID int, allEv []EventInfo) {
+	deg.lock.Lock()
+	defer deg.lock.Unlock()
+	deg.minSeq = firstID
+	deg.allEv = allEv
+}
+
+func TestSequentialAndTrim(t *testing.T) {
```

Comment:
nit: might be worth adding a simple comment to these Test* methods so that is more obvious what they are actually testing.

Reply:
done